### PR TITLE
feat(eslint-plugin): [prefer-as-const] adds support for class properties

### LINKED ISSUE #5405: Bug: [prefer-as-const] Rule doesn't support class properties
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.4&sourceType=module&code=MYGwhgzhAEAKCWAvRZoG8BQ1oAdIBcwAuaAcjwkNOgF4yKqBuDAXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloENktMgHsmyfOii9oI6JHBgAviCVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
class Pizza {
  pasta: 'pasta' = 'pasta';
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-as-const": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Expect an error and a suggestion to use `as const` instead of explicit type.

### Actual Result

No error.

### Additional Info

_No response_

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.31.0` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `X.Y.Z` |

### PR DESCRIPTION
**Pull request by undsoft**
## PR Checklist

-   [x] Addresses an existing open issue: fixes #5405
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

prefer-as-const will now check class members.

### CODE REVIEW

File: packages/eslint-plugin/tests/rules/prefer-as-const.test.ts [hunk 1]
```diff
@@ -25,12 +25,12 @@ ruleTester.run('prefer-as-const', rule, {
     "let foo = 'bar';",
     `
       class foo {
-        bar: 'baz' = 'baz';
+        bar = 'baz';
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/tests/rules/prefer-as-const.test.ts:28]: Now that the rule explicitly handles class properties, we're going to want to test edge cases for them a bit more exhaustively:

* Getters and setters
* Mismatched literal vs. th
* Arrow function with an `as ...`

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Source code looks great, thanks!

Just requesting changes to add a bit more test coverage to class properties.
